File: web/tests/reference/binary-trees/binary-tree-right-side-view.js
function rightSideView(root) {
  if (root === null) return [];
  const out = [];
  let level = [root];
  while (level.length > 0) {
    out.push(level[level.length - 1].val);
    const next = [];
    for (const n of level) {
      if (n.left) next.push(n.left);
      if (n.right) next.push(n.right);
    }
    level = next;
  }
  return out;
}

File: web/tests/reference/binary-trees/binary-tree-right-side-view.py
def right_side_view(root):
    if root is None:
        return []
    out, level = [], [root]
    while level:
        out.append(level[-1].val)
        nxt = []
        for n in level:
            if n.left:
                nxt.append(n.left)
            if n.right:
                nxt.append(n.right)
        level = nxt
    return out
